src/commands/bootstrap/is-folder-empty.ts

```diff
@@ -0,0 +1,72 @@
+// Code copied from `create-next-app`.
+
+import { lstatSync, readdirSync } from 'node:fs';
+import { join } from 'node:path';
+import picocolors from 'picocolors';
+
+// `isFolderEmpty` checks if a folder is empty and ready to onboard a Next.js package into it.
+// It will actually log to stdout as part of its execution.
+export function isFolderEmpty(
+  root: string,
+  name: string,
+  writeStdout: (data: string) => void,
+): boolean {
+  const validFiles = [
+    '.DS_Store',
+    '.git',
+    '.gitattributes',
+    '.gitignore',
+    '.gitlab-ci.yml',
+    '.hg',
+    '.hgcheck',
+    '.hgignore',
+    '.idea',
+    '.npmignore',
+    '.travis.yml',
+    'LICENSE',
+    'Thumbs.db',
+    'docs',
+    'mkdocs.yml',
+    'npm-debug.log',
+    'yarn-debug.log',
+    'yarn-error.log',
+    'yarnrc.yml',
+    '.yarn',
+  ];
```

Comment:
Nitpick, seems like this can be a module-level constant?

Nitpick 2 - maybe we can use a `Set` for it since all we do is check whether an item is in it?

Reply:
I think if it's only used within this function, it should be defined inside this function. (https://kentcdodds.com/blog/colocation)

Agree with making it a Set though, good idea!